src/topp/IDFileConverter.cpp

```diff
@@ -566,6 +585,85 @@ class TOPPIDFileConverter :
         XQuestResultXMLFile().load(in, peptide_identifications, protein_identifications);
       }
 
+      else if (in_type == FileTypes::FASTA)
+      {
+        MSExperiment exp;
+        TheoreticalSpectrumGenerator tsg;
+
+        // extract parameters and remove non tsg params
+        Param p = getParam_().copy("fasta_to_mzml:", true);
+        String enzyme = p.getValue("enzyme");
+        Int mc = p.getValue("missed_cleavages");
+        Int min_charge = p.getValue("min_charge");
+        Int max_charge = p.getValue("max_charge");
+        Int pc_charge = p.getValue("precursor_charge");
+        p.remove("enzyme");
+        p.remove("missed_cleavages");
+        p.remove("min_charge");
+        p.remove("max_charge");
+        p.remove("precursor_charge");
+
+        if (min_charge > max_charge)
+        {
+          writeLog_("Error: 'fasta_to_mzml:min_charge' must be smaller than or equal to 'fasta_to_mzml:max_charge'.");
+          printUsage_();
+          return ILLEGAL_PARAMETERS;
+        }
+
+        if (pc_charge != 0 && pc_charge <= max_charge)
```

Comment:
you can omit this check, right? because the TSG will throw an exception and let the user know

Reply:
Or keep it as an OpenMS precondition. It will only be enabled in debug mode